pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go

```diff
@@ -277,11 +279,37 @@ type addColumnSpec struct {
 	onUpdate *scpb.ColumnOnUpdateExpression
 	comment  *scpb.ColumnComment
 	unique   bool
+	notNull  bool
 }
 
-// addColumn is a helper function which adds column element targets and ensures
-// that the new column is backed by a primary index, which it returns.
+// addColumn adds a column as specified in the `spec`. It delegates most of the work
+// to addColumnIgnoringNotNull and handles NOT NULL constraint itself.
+// It is version aware in that it consults the current active cluster version and set all protobuf
+// fields for version-gating purposes.
 func addColumn(b BuildCtx, spec addColumnSpec, n tree.NodeFormatter) (backing *scpb.PrimaryIndex) {
```

Comment:
You could say more here about what the version gate is all about, why it exists, how long it should last etc. If you stumbled upon this code and had not written it, would you know what it meant?